### DISCUSSION TITLE
close window by ESCAPE

### DISCUSSION THREAD

**Discussion by forrestjgq** [isl-org/Open3D#4597] [General]
@errissa I saw ESCAPE will close the window by https://github.com/isl-org/Open3D/pull/3992

Would you consider more about that action? As a platform, it might be reasonable to let the user decide the interaction like capture the key event and close window manually.

I notice that issue because I migrate my project from 0.13 to 0.14 and all ESCAPE I used as shortcut is not working now.

**Comment by joseph-sch** [comment]
I second this request. When I open a dialog, if the user presses Escape with the aim of closing the dialog, it closes the whole application.

I tried to catch the Escape key and I see that the callback set with `Window.set_on_key` _is_ called when the Escape key is pressed, but whether the callback return `True` or `False`, the application closes after the callback returns.

**Reply by lee-deraud** [reply]
(FWIW, I submitted a feature request (#7436) on this very topic, not having seen this discussion or #3992 ...sorry about that.)
I discovered that the escape key behavior is *very* odd. I can trap the keystroke (both up and down), no problem. From my key event handler:
```
            elif k == gui.KeyName.ESCAPE.value:
                print("Escape key",flush=True)
                self.escape_hit = True
                return(gui.Widget.EventCallbackResult.CONSUMED)
```
and I can trap the main window 'close' event, which I thought would solve the problem:
```
    def on_close(self):
        print("On close, esc = ",self.escape_hit,flush=True)
        if self.escape_hit:
            self.escape_hit = False
            return False
        return True
```
But if escape key is hit, I see this:
```
On close, esc =  False
On close, esc =  False
Escape key
Escape key
Exiting...
```
Clicking the 'X' (close icon) in the upper-right just gives this:
```
On close, esc =  False
Exiting...
```
Some kind of race condition in the event handling?
